listenbrainz/webserver/views/atom.py

```diff
@@ -0,0 +1,570 @@
+    :statuscode 200: The feed was successfully generated.
+    :statuscode 400: Bad request.
+    :statuscode 404: The user does not exist.
+    :resheader Content-Type: *application/atom+xml*
+    """
+    user = db_user.get_by_mb_id(db_conn, user_name)
+    if user is None:
+        return Response(status=404)
+
+    minutes = request.args.get("minutes", DEFAULT_MINUTES_OF_LISTENS)
+    if minutes:
+        try:
+            minutes = int(minutes)
+        except ValueError:
+            return Response(status=400)
+    if minutes < 1 or minutes > MAX_MINUTES_OF_LISTENS:
+        return Response(status=400)
+
+    # estimate limit from minutes, assuming 1 listen per minute.
+    # mostly likely there won't be that much listens, but with such padding
+    # it's less likely for feed readers to miss listens.
+    limit = minutes
+
+    to_ts = datetime.now()
+    listens, _, _ = timescale_connection._ts.fetch_listens(
+        user, to_ts=to_ts, limit=limit
+    )
+
+    fg = _init_feed(
+        _external_url_for(".get_listens", user_name=user_name),
+        f"Listens for {user_name} - ListenBrainz",
+        _external_url_for("user.index", path="", user_name=user_name),
+        _external_url_for(".get_listens", user_name=user_name),
+    )
+
+    # newer listen comes first
+    for listen in reversed(listens):
+        track_name = listen.data["track_name"]
+        recording_mbid = listen.data["additional_info"].get("recording_mbid")
+        artist_name = listen.data["artist_name"]
+        artist_mbid = (
+            listen.data["additional_info"].get("artist_mbids")[0]
+            if listen.data["additional_info"].get("artist_mbids")
+            else None
+        )
+
+        fe = fg.add_entry()
+        # according to spec, ids don't have to be deferencable.
+        fe.id(
+            f"{_external_url_for('.get_listens', user_name=user_name)}/{listen.ts_since_epoch}/{track_name}"
+        )
+        fe.title(f"{track_name} - {artist_name}")
+
+        _content = render_template(
+            "atom/listens.html",
+            user={
+                "href": _external_url_for("user.index", path="", user_name=user_name),
+                "user_name": user_name,
+            },
+            track={
+                "href": f"https://musicbrainz.org/recording/{recording_mbid}",
+                "track_name": track_name,
+            },
+            artist={
+                "href": _external_url_for("artist.artist_page", path=artist_mbid),
+                "artist_name": artist_name,
+            },
+        )
+        fe.content(
+            content=_content,
+            type="html",
+        )
+
+        fe.published(listen.timestamp)
+        fe.updated(listen.timestamp)
+
+    atomfeed = fg.atom_str(pretty=True)
+
+    return Response(atomfeed, mimetype="application/atom+xml")
+
+
+@atom_bp.route("/fresh_releases", methods=["GET"])
```

Comment:
Please change the URLs to use -, rather than _ for all URLs. More consistent with the rest of our URLs.